presets/ragengine/embedding/huggingface_local_embedding.py

```diff
@@ -1,12 +1,46 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
-from typing import Any
+from typing import Any, List, Union
+import time
+from io import BytesIO
+from sentence_transformers import SentenceTransformer
 from llama_index.embeddings.huggingface import HuggingFaceEmbedding
 from .base import BaseEmbeddingModel
+from ragengine.metrics.helpers import record_embedding_metrics
+from ragengine.metrics.prometheus_metrics import (
+    rag_embedding_requests_total, 
+    rag_embedding_latency,
+    STATUS_SUCCESS,
+    STATUS_FAILURE,
+    MODE_LOCAL
+)
 
 class LocalHuggingFaceEmbedding(HuggingFaceEmbedding, BaseEmbeddingModel):
+    """HuggingFace embedding model with metrics collection."""
+    
+    def _embed_with_retry(
+        self,
+        inputs: List[Union[str, BytesIO]],
+        prompt_name: Any = None,
+    ) -> List[List[float]]:
+        start_time = time.time()
+        status = STATUS_SUCCESS
+        
+        try:
+            result = super()._embed_with_retry(inputs, prompt_name)
+            return result
+        except Exception as e:
+            status = STATUS_FAILURE
+            raise e
```

Comment:
```suggestion
            raise
```
python preserves traceback